zeeguu/core/word_scheduling/basicSR/basicSR.py

```diff
@@ -52,29 +58,36 @@ def update_schedule(self, db_session, correctness):
                 db.session.commit()
                 return
 
-            if datetime.now() < self.next_practice_time:
+            # Use the same logic as when selecting bookmarks
+            # Avoid case where if schedule at 01-01-2024 11:00 and user does it at
+            # 01-01-2024 10:00 the status is not updated.
+            if self.get_current_study_window() < self.next_practice_time:
                 # a user might have arrived here by doing the
                 # bookmarks in a text for a second time...
                 # in general, as long as they didn't wait for the
                 # cooldown perio, they might have arrived to do
                 # the exercise again; but it should not count
                 return
-
-            new_cooling_interval = NEXT_COOLING_INTERVAL_ON_SUCCESS[
-                self.cooling_interval
-            ]
-            next_practice_date = datetime.now() + timedelta(
-                minutes=new_cooling_interval
+            # Since we can now loose the streak on day 8,
+            # we might have to repeat it a few times to learn it.
+            new_cooling_interval = NEXT_COOLING_INTERVAL_ON_SUCCESS.get(
+                self.cooling_interval, 8 * ONE_DAY
```

Comment:
It's not readily clear to me why the `.get.(..., 8 days)` pattern? i guess if we're on day 8 we get another day 8 interval?

Reply:
The reason was because if we keep track of the consecutive answers, I would then just schedule the word every 8 days until the user gets it correct. But I will remove the consecutive number of answers.